internal/local/structure_v2.go

```diff
@@ -331,3 +327,204 @@ func parseSync(rootDir string) (*SyncStructure, error) {
+		return err
+	}
+	if err := writeSync(rootDir, a.Sync); err != nil {
+		return err
+	}
+	if err := writeDataSources(rootDir, a.DataSources); err != nil {
+		return err
+	}
+	if err := writeHTTPEndpoints(rootDir, a.HTTPEndpoints); err != nil {
+		return err
+	}
+	return nil
+}
+
+func writeFunctionsV2(rootDir string, functions *FunctionsStructure) error {
+	configs := []map[string]interface{}{}
+	if functions != nil {
+		configs = functions.Configs
+	}
+	dataFuncs, err := MarshalJSON(configs)
+	if err != nil {
+		return err
+	}
+	err = WriteFile(filepath.Join(rootDir, NameFunctions, FileConfig.String()), 0666, bytes.NewReader(dataFuncs))
+	if err != nil {
+		return err
+	}
+	if functions == nil {
+		return nil
+	}
+	for path, src := range functions.Sources {
+		err = WriteFile(filepath.Join(rootDir, NameFunctions, path), 0666, bytes.NewReader([]byte(src)))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func writeAuth(rootDir string, auth *AuthStructure) error {
+	if auth == nil {
+		return nil
+	}
+	if auth.Providers != nil {
+		dataProvider, err := MarshalJSON(auth.Providers)
+		if err != nil {
+			return err
+		}
+		err = WriteFile(filepath.Join(rootDir, NameAuth, FileProviders.String()), 0666, bytes.NewReader(dataProvider))
+		if err != nil {
+			return err
+		}
+	}
+	if auth.CustomUserData != nil {
+		dataCustomUser, err := MarshalJSON(auth.CustomUserData)
+		if err != nil {
+			return err
+		}
+		err = WriteFile(filepath.Join(rootDir, NameAuth, FileCustomUserData.String()), 0666, bytes.NewReader(dataCustomUser))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func writeSync(rootDir string, sync *SyncStructure) error {
+	if sync == nil || sync.Config == nil {
+		return nil
+	}
+	data, err := MarshalJSON(sync.Config)
+	if err != nil {
+		return err
+	}
+	err = WriteFile(filepath.Join(rootDir, NameSync, FileConfig.String()), 0666, bytes.NewReader(data))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func writeDataSources(rootDir string, dataSources []DataSourceStructure) error {
+	if err := os.MkdirAll(filepath.Join(rootDir, NameDataSources), os.ModePerm); err != nil {
+		return err
+	}
+	for _, ds := range dataSources {
+		nameDataSource := ds.Config["name"].(string)
+		dataDataSource, err := MarshalJSON(ds.Config)
+		if err != nil {
+			return err
+		}
+		err = WriteFile(filepath.Join(rootDir, NameDataSources, nameDataSource, FileConfig.String()), 0666, bytes.NewReader(dataDataSource))
+		if err != nil {
+			return err
+		}
+		for _, rule := range ds.Rules {
+			schema := rule[NameSchema]
+			dataSchema, err := MarshalJSON(schema)
+			if err != nil {
+				return err
+			}
+			delete(rule, NameSchema)
+			dataRule, err := MarshalJSON(rule)
+			if err != nil {
+				return err
+			}
+			err = WriteFile(filepath.Join(rootDir, NameDataSources, nameDataSource, fmt.Sprintf("%s", rule["database"]), fmt.Sprintf("%s", rule["collection"]), FileRules.String()), 0666, bytes.NewReader(dataRule))
```

Comment:
should probably consolidate the logic for these filepaths here too, common `dir` vars and whatnot